src/it/scala/net/snowflake/spark/snowflake/SnowflakeResultSetRDDSuite.scala

```diff
@@ -1496,9 +1498,252 @@ class SnowflakeResultSetRDDSuite extends IntegrationSuiteBase {
+    val partitionCount = 1
+    val rowCountPerPartition = 1024 * 50
+    // It is enough to run this test on AWS for Arrow
+    if (Option(System.getenv("SNOWFLAKE_TEST_ACCOUNT")).getOrElse("aws").equals("aws") && !params.useCopyUnload) {
+      val thisSparkSession = SparkSession
+        .builder()
+        .appName("Spark SQL basic example")
+        .config("spark.some.config.option", "some-value")
+        .config("spark.master", "local")
+        .getOrCreate()
+
+      def getRandomString(len: Int): String = {
+        Random.alphanumeric take len mkString ""
+      }
+
+      // Create RDD which generates 1 large partition
+      val testRDD: RDD[Row] = thisSparkSession.sparkContext
+        .parallelize(Seq[Int](), partitionCount)
+        .mapPartitions { _ => {
+          (1 to rowCountPerPartition).map { _ => {
+            // generate value for each row for large compressed size
+            val strValue = getRandomString(512)
+            Row(strValue, strValue, strValue, strValue)
+          }
+          }.iterator
+        }
+        }
+      val schema = StructType(
+        List(
+          StructField("str1", StringType),
+          StructField("str2", StringType),
+          StructField("str3", StringType),
+          StructField("str4", StringType)
+        )
+      )
+
+      // Convert RDD to DataFrame
+      val df = thisSparkSession.createDataFrame(testRDD, schema)
+
+      // Write to snowflake with upload multiple part API
+      jdbcUpdate(s"drop table if exists $test_table_write")
+      df.write
+        .format(SNOWFLAKE_SOURCE_NAME)
+        .options(thisConnectorOptionsNoTable)
+        .option(Parameters.PARAM_UPLOAD_CHUNK_SIZE_IN_MB, "5")
+        .option("dbtable", test_table_write)
+        .mode(SaveMode.Overwrite)
+        .save()
+      assert(getRowCount(test_table_write) == partitionCount * rowCountPerPartition)
+
+      // Write to snowflake without upload multiple part API
+      jdbcUpdate(s"drop table if exists $test_table_write")
+      df.write
+        .format(SNOWFLAKE_SOURCE_NAME)
+        .options(thisConnectorOptionsNoTable)
+        .option(Parameters.PARAM_USE_AWS_MULTIPLE_PARTS_UPLOAD, "off")
+        .option("dbtable", test_table_write)
         .mode(SaveMode.Overwrite)
         .save()
+      assert(getRowCount(test_table_write) == partitionCount * rowCountPerPartition)
+
+      // Negative test to inject exception after upload 2nd block
+      TestHook.enableTestFlagOnly(TestHookFlag.TH_FAIL_UPLOAD_AWS_2ND_BLOCK)
+      assertThrows[Exception]({
+        df.write
+          .format(SNOWFLAKE_SOURCE_NAME)
+          .options(thisConnectorOptionsNoTable)
+          .option(Parameters.PARAM_UPLOAD_CHUNK_SIZE_IN_MB, "5")
+          .option("dbtable", test_table_write)
+          .mode(SaveMode.Overwrite)
+          .save()
+      })
+      TestHook.disableTestHook()
+    }
+  }
+
+  ignore("perf test for AWS multiple parts upload API") {
```

Comment:
why we ignore this test?

Reply:
This test case is used to evaluate the performance with old API and new API with different buffer size. It's more like a manual test and it needs a long time time to run. So it is not necessary to enable it.

Reply:
And this test indicates there is no perf difference with old API and new API with different buffer size.